lib/matplotlib/backend_tools.py

```diff
@@ -99,20 +98,6 @@ def trigger(self, sender, event, data=None):
 
         pass
 
-    @figure.setter
-    def figure(self, figure):
-        """
-        Set the figure
-
-        Set the figure to be affected by this tool
-
-        Parameters
-        ----------
-        figure: `Figure`
-        """
-
-        self._figure = figure
```

Comment:
What I was thinking, and I really think is the best way to do it, is to emit an event from FigureManager when the figure gets setted.
The tools that need to refresh information based on the figure will hook into that event, and store the figures in a local array or dictionary.
Having a reference to figure here makes me nervous.